### PR TITLE
Pvs cleanup

### PR DESCRIPTION
**Pull request by allenh1**
This cleans up a good amount of the PVS studio detected errors.

New PVS log is [here](http://git.allen-software.com:8080/job/allenh1/job/YaSh/job/pvs-cleanup/PVS-Studio_20Report/).

### CODE REVIEW

**Review by mechaHarry — APPROVED** [review]
Logical equivalencies, syntax restructures, and pvs optimizations/patches all look great to me.

Lock n load baby,
- mH

File: src/wildcard.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@ void wildcard_expand(const std::shared_ptr<char> arg)
   char * a = arg.get();
 
   for (int i = 0; *(a + 1); a++, i++) {
-    if ((*a == '/' && *(a + 1) == '.' && *(a + 1) == '*') ||
+    if ((*a == '/' && (*(a + 1) == '.' || *(a + 1) == '*')) ||
```
Comment (mechaHarry) [inline src/wildcard.cpp]: This might be the root cause for travis CI failure

**Review by mechaHarry — CHANGES_REQUESTED** [review]

File: src/wildcard.cpp (hunk 1 above)
Comment (mechaHarry) [reply src/wildcard.cpp]: it isn't; this piece of code is resolving hiddens fine, somewhere else is not removing dot-files from the output when echoing.

**Review by mechaHarry — APPROVED** [review]

**Comment by mechaHarry** [comment]
Seemed to fix the expected wildcarding output via latest push, but the pr still seems to miss one of the zombie cases; if we can resolve that, then this pr is ready to launch.

**Comment by allenh1** [comment]
> Seemed to fix the expected wildcarding output via latest push, but the pr still seems to miss one of the zombie cases

Yeah, that test has always been flakey. If it fails several times in a  row I start to worry, but it usually works with a re-trigger.

Also, in the future, please refrain from adding commits to my PR... I squash merge, so your commits will all be creddited to me and you deserve credits!

Since the CI is green now, I'm going to merge this in.

**Pull request merged**